Branch-SDK/src/io/branch/search/RecommendationBuilder.java

```diff
@@ -0,0 +1,165 @@
+    private static final int DEF_MAX_CONTENT_REC_COUNT = 3;
+    private int maxAppRecommendationCount;
+    private int maxContentRecommendationCount;
+    private boolean skipSystemApps;
+    private final List<String> preferredAppsForContents;
+    private final IRecommendationEvents callback;
+
+    /**
+     * <p>
+     * Builder for getting content and app recommendations from Branch.
+     * Branch will provide content app recommendation based on the configuration. The
+     * contents and apps are ranked with recent interactions and other contextual parameters
+     * </p>
+     *
+     * @param callback {@link io.branch.search.RecommendationBuilder.IRecommendationEvents} instance to listen to recommendation events
+     */
+    public RecommendationBuilder(IRecommendationEvents callback) {
+        this.callback = callback;
+        this.maxAppRecommendationCount = DEF_MAX_APP_REC_COUNT;
+        this.maxContentRecommendationCount = DEF_MAX_CONTENT_REC_COUNT;
+        this.skipSystemApps = true;
+        this.preferredAppsForContents = new ArrayList<>();
+    }
+
+    /**
+     * (
+     * Set the maximum number of recommended apps in the recommendations
+     *
+     * @param maxAppRecommendations Max num of apps in the recommended apps list
+     * @return RecommendationBuilder instance for chaining
+     */
+    public RecommendationBuilder setMaxAppRecommendations(int maxAppRecommendations) {
+        this.maxAppRecommendationCount = maxAppRecommendations;
+        return this;
+    }
+
+    /**
+     * (
+     * Set the maximum number of recommended contents in the recommendations
+     *
+     * @param maxContentRecommendations Max num of contents in the recommended content list
+     * @return RecommendationBuilder instance for chaining
+     */
+    public RecommendationBuilder setMaxContentRecommendations(int maxContentRecommendations) {
+        this.maxContentRecommendationCount = maxContentRecommendations;
+        return this;
+    }
+
+    /**
+     * Setting this will skip all system apps from apps and content recommendation list
+     *
+     * @return RecommendationBuilder instance for chaining
+     * @param  isSkipSystemApps {@code true} to skip the systems apps from app and their contents from recommendation
+     */
+    public RecommendationBuilder skipSystemApps(boolean isSkipSystemApps) {
+        this.skipSystemApps = isSkipSystemApps;
+        return this;
+    }
+
+    /**
+     * Add the given app to the app preference list. Caller can add any number of preferred apps
+     * The recommended content list will include only contents from the preferred apps.
+     * If no apps added, the content recommendations are left unfiltered with apps
+     *
+     * @param packageName {@link String} package name for the apps to be added to the preferred list
+     * @return RecommendationBuilder instance for chaining
+     */
+    public RecommendationBuilder addPreferredApp(String packageName) {
```

Comment:
preferred apps provide a a filtering for results in recommendation. If preference added only  contents from those apps will appear in th recomemndation